### PR TITLE
Create convert_day.py

### PR DESCRIPTION
**Pull request by HieuTam**
_Dear @capigiba and all concerned,_

The `convert_day.py` module provides four functions: `IsLeapYear(year)`, `ConvertDayToTimestamp(day, month, year)`, `GetMonthRange(month, year)`, and `GetYearRange(year)`.

1. `IsLeapYear(year)` determines if a year is a leap year based on divisibility rules: divisible by 4 but not 100, unless also divisible by 400.

2. `ConvertDayToTimestamp(day, month, year)` converts a date (e.g., 01/01/2025) to a Unix timestamp (e.g., 1735743797) with a default time of 00:00:00. It checks:
   - If inputs are integers; otherwise, it returns 0 and an error message.
   - If the `year`, `month`, and `day` are valid for the given date, based on leap year rules.
   - If valid, it returns the timestamp and `None` for errors.

3. `GetMonthRange(month, year)` calculates the Unix timestamps for the start and end of a month:
   - Validates `month` and `year` as integers.
   - Uses `ConvertDayToTimestamp` to get the start of the month. If invalid, returns an error.
   - Calculates the end of the month by finding the start of the next month, then subtracting 1 second to get the last moment of the current month.
   - Returns the start, end timestamps, and any error messages.

4. `GetYearRange(year)` calculates the Unix timestamps for the start and end of a year:
   - Validates `year` as an integer.
   - Gets the start of the year using `ConvertDayToTimestamp`. If invalid, returns an error.
   - Finds the end of the year by calculating the start of the next year and subtracting 1 second.
   - Returns the start, end timestamps, and any error messages.

The test cases check the accuracy and stability of the `convert_day.py` functions, ensuring they return correct Unix timestamps for valid inputs, handle invalid inputs (e.g., non-integers, out-of-range dates) with appropriate errors, and manage edge cases like leap years without crashing. The attached screenshot shows test scenarios for both valid and invalid inputs.

image
image

_Best regards,_
Hita.

### CODE REVIEW

File: convert_day.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+from datetime import datetime
+
+def IsLeapYear(year):
```
Comment (capigiba) [inline convert_day.py:3]: As we discussed earlier, if this is a supporting function, it means it won’t appear anywhere else except in this file. So this time, you should name it `isLeapYear` to help distinguish it from functions that are called externally

Comment (capigiba) [inline convert_day.py:3]: When coding, clarity is something we need to prioritize. This function is good, but there’s room for improvement to make it better.

For example, currently, you’ve written:
`def isLeapYear(year):`
This can be confusing during code review because it’s unclear what type of data you’re passing and what the function returns. Instead, you could write:
`def is_leap_year(year: int) -> bool`

With this, it’s much easier to understand the input and output of the function. Additionally, this creates a safeguard for the function. Imagine if someone accidentally passed a boolean value as `year`; the function would crash. Adding type hints can help prevent such issues

**Comment by capigiba** [comment]
The overall logic looks great now. The remaining task before moving on is to clarify those functions based on my review

File: convert_day.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
+from datetime import datetime
+
+def isLeapYear(year: int) -> bool:
+    """Check if a year is a leap year."""
+    return (year % 4 == 0) and ((year % 100 != 0) or (year % 400 == 0))
+
+def ConvertDayToTimestamp(day, month, year):
```
Comment (capigiba) [inline convert_day.py:7]: you should add type in here

File: convert_day.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    elif (month == 2) and (day > 28):
+        return 0, f"Invalid input day. February {year} (Common year) only has 28 days." 
+    
+    dt = datetime(year, month, day, 0, 0) 
+    unixTimestamp = int(dt.timestamp())
+    err = None
+
+    return unixTimestamp, err
+
+def GetMonthRange(month, year):
```
Comment (capigiba) [inline convert_day.py:41]: the same with this

File: convert_day.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        endOfMonth, err2 = ConvertDayToTimestamp(1, 1, year + 1)
+    else:
+        endOfMonth, err2 = ConvertDayToTimestamp(1, month + 1, year)
+    endOfMonth -= 1 if endOfMonth > 0 else 0 # Adjust to the end of the month (from 00:00:00 of 01/mm + 1/yyyy to 23:59:59 of dd/mm/yyyy)
+
+    err = err1 if err1 == err2 else err2 or err1 or f"{err1}; {err2}"
+
+    return startOfMonth, endOfMonth, err
+
+def GetYearRange(year):
```
Comment (capigiba) [inline convert_day.py:69]: here too :)
